src/displayapp/screens/StopWatch.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include "Screen.h"
+#include "components/datetime/DateTimeController.h"
+#include "../LittleVgl.h"
+
+#include "FreeRTOS.h"
+#include "portmacro_cmsis.h"
+
+#include <array>
+
+namespace Pinetime::Applications::Screens {
+
+  enum class States { INIT, RUNNING, HALTED };
+
+  enum class Events { PLAY, PAUSE, STOP };
+
+  struct TimeSeparated_t {
+    int mins;
+    int secs;
+    int msecs;
+  };
+
+  // A simple buffer to hold the latest two laps
+  template <int N> struct LapTextBuffer_t {
+    LapTextBuffer_t() : _arr {}, currentSz {}, capacity {N}, head {-1} {
+    }
+
+    void addLaps(const TimeSeparated_t& timeVal) {
+      head++;
+      head %= capacity;
+      _arr[head] = timeVal;
+
+      if (currentSz < capacity) {
+        currentSz++;
+      }
+    }
+
+    void clearBuffer() {
+      _arr = {};
+      currentSz = 0;
+      head = -1;
+    }
+
+    TimeSeparated_t* operator[](std::size_t idx) {
+      // Sanity check for out-of-bounds
+      if (idx >= 0 && idx < capacity) {
+        if (idx < currentSz) {
+          // This transformation is to ensure that head is always pointing to index 0.
+          const auto transformed_idx = (head - idx) % capacity;
+          return (&_arr[transformed_idx]);
+        }
+      }
+      return nullptr;
+    }
+
+  private:
+    std::array<TimeSeparated_t, N> _arr;
```

Comment:
Also for more consistency with the rest of the code : could you rename `_arr` as `array' for example? And maybe also `currentSz` to `currentSize`?